File: BegginersSelection/ABC086C.rb
n = gets.chomp.to_i
t = [0, 0]
x = [0, 0]
y = [0, 0]

can = true
(1..n).each do |i|
    line = gets.chomp.split(" ").map(&:to_i)
    t = [t[1], line[0]]
    x = [x[1], line[1]]
    y = [y[1], line[2]]
    dt = t[1]-t[0]
    dist = (x[1]-x[0]).abs + (y[1]-y[0]).abs
    if dt < dist
        can = false
    elsif dt % 2 != dist % 2
        can = false
    end
end

if can
    puts "Yes"
else
    puts "No"
end

File: BegginersSelection/Golang/ABC086C.go
package main

import "fmt"

func main() {
	var n int
	fmt.Scan(&n)
	t := make([]int, n+1)
	x := make([]int, n+1)
	y := make([]int, n+1)
	t[0] = 0
	x[0] = 0
	y[0] = 0
	for i := 1; i < n+1; i++ {
		fmt.Scan(&t[i], &x[i], &y[i])
	}

	can := true

	for i := 0; i < n; i++ {
		dt := t[i+1] - t[i]
		dist := (x[i+1] - x[i]) + (y[i+1] - y[i])
		if dt < dist {
			can = false
			break
		}
		if dif := dt - dist; dif%2 == 1 {
			can = false
			break
		}
	}

	if can {
		fmt.Println("Yes")
	} else {
		fmt.Println("No")
	}
}
